### PR TITLE
First draft of Javascript guide

### PR DESCRIPTION
**Pull request by jeremiak**
I have changed the names of the files and URLs from using "node" to using "javascript" to stick with the other languages convention.

I've also gone ahead and put in the content from the Google doc.

Hey JS folks like @dhcole and @msecret - would you mind taking a look? The goal of these guides is to make it a bit less scary to have to set up your environment with a new language.

### CODE REVIEW

**Comment by mbland** [comment]
Hey @jeremiak, don't forget to run `./go update_nav` and then hand-edit the `navigation:` section in `_config.yml` to get the sidebar to show up correctly.

File: pages/languages/javascript.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+## Node Installation
+
+Node is super easy to install and you can do so in a number of ways. The fine folks behind the project have put a bunch of time and energy into ensuring cross platform compatibility and it totally shows.
+
+You can install it through the binaries on nodejs.org, through `brew` on Mac OS X, or compiling it from source but the preferred method is through a tool called `nvm`. `nvm` stands for Node Version Manager. It will enable us to easily install and use different versions of Node (including the relatively short lived iojs fork).
+
+## nvm Installation
+
+You can find the homepage for the project at https://github.com/creationix/nvm and [installation instructions](https://github.com/creationix/nvm#install-script) there. Overall, it is a really easy tool to set up, and can generally done with copying and pasting the following line of code into your terminal:
+curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.29.0/install.sh | bash
```
Comment (msecret) [inline pages/languages/javascript.md]: I think we should have a paragraph or at least a callout to the windows version of nvm. I'd go with this one: https://github.com/coreybutler/nvm-windows as it seems easier to setup. I think the use instructions are the same too.

Based on my understanding of this repo, supporting Windows would be very important, right?

Comment (jeremiak) [inline pages/languages/javascript.md]: Great point, Marco. I think we can add a reference to installing on Windows and point to that library. It looks pretty good and was updated somewhat recently.

**Comment by jeremiak** [comment]
Will do, @mbland.

We should hold off on merge this PR until we add a section for `n` because that's what we're [installing with the laptop script](https://github.com/18F/laptop/blob/master/mac#L170). I know @rogeruiz uses `n` so maybe he can contribute something quick? Roger, these guides are meant for folks (likely coming from other languages) to get their local environment set up. 

Also, a resulting question is to what extent should the guidance in these guides and the installed packages in the laptop script line up? I would imagine that it would be important for them to be as similar as possible because part of the goal of these guides is to make setting up environments easier.

However, `nvm` and `n` are so similar that I think perhaps including them both would be ok. Thoughts?

**Comment by rogeruiz** [comment]
From my understanding of `n` versus `nvm` is that `n` will only be able to install binaries from node distributions while some older versions may not be available so they'll need to be downloaded, compiled, and manually moved to the directory that `n` uses to manage it's node versions. The default directory is `/usr/local/n/versions` but can be changed. [The docs talk about that here](https://github.com/tj/n#installation).

The benefits I find between `n` and `nvm` is that `nvm` will install your node versions into the `$HOME` directory, so some IDEs or Text Editors ( WebStorm, et all ) that don't normally read the `$PATH` variable will need to have a direct path to `node` ( i.e. `~/n/bin/node/path` ) in order to use it. In my opinion the better way is how `n` will always copy the files from `/usr/local/n/versions` to `/usr/local/bin/node` when selecting your node versions. The approach that `n` takes sits better with me since it doesn't need to manipulate your `$PATH` variable at all.

While these two approaches are different, having both of them in the docs might be a good idea. Just include the caveat about having to link directly to your `node` path within certain editors or IDEs or to  check your `$PATH` variable using `echo $PATH` in your terminal to make sure that the `nvm` path is set properly.

I can write something up about that caveat, or steps to troubleshoot it, if needed and maintain it on this PR. Let me know.

**Comment by msecret** [comment]
@rogeruiz does `n` require you to use `sudo` when doing actions which manipulate the node environment. For instance, do you have to use `sudo`when installing something globally with `npm`?

**Comment by rogeruiz** [comment]
@msecret it doesn't as long as the user running the command already has ownership of `/usr/local` which I recommend since most of our boxes are single user machines anyways.

**Comment by mbland** [comment]
Gonna go ahead and merge so we've got something to work with. All of this is a living doc, so feel free to continue to discuss and open new issues/PRs.

**Pull request merged**